src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -222,8 +225,13 @@ def construct_payload(self):
 
             # Check if updating existing container
             updating_existing_container = False
+            if self.get_argument_force_single_container_updates():
+                # Remove n-1 containers where n is the number of containers in the containerapp and replace remaining container with the new container
+                # Fails if all containers are removed
+                while (len(self.new_containerapp["properties"]["template"]["containers"]) > 1):
+                    self.new_containerapp["properties"]["template"]["containers"].pop()
             for c in self.new_containerapp["properties"]["template"]["containers"]:
-                if c["name"].lower() == self.get_argument_container_name().lower():
+                if c["name"].lower() == self.get_argument_container_name().lower() or self.get_argument_force_single_container_updates():
```

Comment:
Can you add some description about this change in here about this argument?

Reply:
Added a comment about this argument in my latest commit.